twonav/fixes/fix_unified_kernel_control_files.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+KERNEL_SRC_PATH=$1
+
+readonly BRANDS="twonav os"
+readonly MODELS="trail aventura"
+readonly KERN_IMAGE_BASE_NAME="linux-image-4.1.15"
+readonly KERN_HEADERS_BASE_NAME="linux-headers-4.1.15"
+readonly KERNEL_NAME=twonav-2018
+readonly KERNEL_PRINT="$(awk -F '-' '{print $3}'<<<$KERN_IMAGE_BASE_NAME)-$KERNEL_NAME"
+readonly BREAK_VERSION="1.0.19"
+
+function pkg_info_gen() {
+	echo -n "$1"
+	local i j cnt=0
+	for i in $BRANDS ; do for j in $MODELS ; do
+		if ((cnt++ > 0)) ; then echo "," ; fi
+		echo -en "\t$2-$i-$j-2018 (<= $BREAK_VERSION)"
+	done ; done
+	echo
+}
+
+function fix_unified_kernel_control_files() {
+	local i
+	readonly LINUX_IMAGE_DEBIAN_PATH=${1}/debian/$KERN_IMAGE_BASE_NAME-$KERNEL_NAME/DEBIAN
+	readonly LINUX_HEADERS_DEBIAN_PATH=${1}/debian/$KERN_HEADERS_BASE_NAME-$KERNEL_NAME/DEBIAN
+
+	for i in "Breaks:" "Replaces:" ; do
+		pkg_info_gen "$i" $KERN_IMAGE_BASE_NAME >> $LINUX_IMAGE_DEBIAN_PATH/control
+		pkg_info_gen "$i" $KERN_HEADERS_BASE_NAME >> $LINUX_HEADERS_DEBIAN_PATH/control
+	done
+}
+
+fix_unified_kernel_control_files $KERNEL_SRC_PATH
+
```

Comment:
@lluisdiaz for avoiding having two "Breaks" and two "Replaces", e.g.

#!/bin/bash                                                                     
                                                                                
KERNEL_SRC_PATH=$1                                                              
                                                                                
readonly BRANDS="twonav os"                                                     
readonly MODELS="trail aventura"                                                
readonly KERN_IMAGE_BASE_NAME="linux-image-4.1.15"                              
readonly KERN_HEADERS_BASE_NAME="linux-headers-4.1.15"                          
readonly KERNEL_NAME=twonav-2018                                                
readonly KERNEL_PRINT="$(awk -F '-' '{print $3}'<<<$KERN_IMAGE_BASE_NAME)-$KERNEL_NAME"
readonly BREAK_VERSION="1.0.19"                                                 
                                                                                
function pkg_info_gen() {                                                       
        local SECTION=$1 BASE_LIST="${@:2}"                                     
        echo -n "$SECTION"                                                      
        local i j k cnt=0                                                       
        for i in $BASE_LIST ; do for j in $BRANDS ; do for k in $MODELS ; do    
                if ((cnt++ > 0)) ; then echo "," ; fi                           
                echo -en "\t$i-$j-$k-2018 (<= $BREAK_VERSION)"                  
        done ; done ; done                                                      
        echo                                                                    
}                                                                               
                                                                                
function fix_unified_kernel_control_files() {                                   
        local i                                                                 
        readonly LINUX_IMAGE_DEBIAN_PATH=${1}/debian/$KERN_IMAGE_BASE_NAME-$KERNEL_NAME/DEBIAN
        readonly LINUX_HEADERS_DEBIAN_PATH=${1}/debian/$KERN_HEADERS_BASE_NAME-$KERNEL_NAME/DEBIAN
                                                                                
        for i in "Breaks:" "Replaces:" ; do                                     
                pkg_info_gen "$i" $KERN_IMAGE_BASE_NAME \                       
                                  $KERN_HEADERS_BASE_NAME \                     
                                        >> $LINUX_IMAGE_DEBIAN_PATH/control     
        done                                                                    
}                                                                               
                                                                                
fix_unified_kernel_control_files $KERNEL_SRC_PATH

Reply:
@faragon2n , si he entendido bien, ¿propones que el nuevo "image" genérico haga el Breaks / Replaces de todos los "headers" y "image" específicos?

No tengo claro que eso funcione, ya que el nuevo "headers" genérico tiene componentes en común con los "headers" específicos, al igual que el nuevo "image" genérico los tiene en común con los específicos.

Reply:
mmm I think it's more confusing the way 2 files are treatten in the same loop.

Reply:
my fault, I missed they were going to different files